src/api.js

```diff
@@ -49,13 +68,38 @@ const getCsp = async (siteName) => {
     return await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/netlify-toml`);
 }
 
+const createPageData = async ({folderName, subfolderName, newFileName, siteName, resourceName}, content) => {
+    const apiEndpoint = getPageApiEndpoint({folderName, subfolderName, newFileName, siteName, resourceName})
+    const params = { content }
+    await axios.post(apiEndpoint, params)
+    
+    // redirect to new page upon successful creation
+    if (folderName) {
+        return `/sites/${siteName}/folder/${folderName}/${encodeURIComponent(`${subfolderName ? `${subfolderName}/` : ''}${newFileName}`)}`
+    } 
+    if (resourceName) {
+        return `/sites/${siteName}/resources/${resourceName}/${encodeURIComponent(newFileName)}`
+    }
+    return `/sites/${siteName}/pages/${encodeURIComponent(newFileName)}`
```

Comment:
We shouldn't be encoding the URI component here, since this is the redirection URL, which is handled appropriately by our router. We also use a different url for subfolders - we can substitute this chunk with

```    
if (folderName) {
        return `/sites/${siteName}/folder/${folderName}/${subfolderName ? `subfolder/${subfolderName}/` : ''}${newFileName}`
    } 
    if (resourceName) {
        return `/sites/${siteName}/resources/${resourceName}/${newFileName}`
    }
    return `/sites/${siteName}/pages/${newFileName}`
```

Reply:
thanks for pointing it out! fixed 83aa2f4